### PR TITLE
impl InputSource impl for TransactionsRecordsById

### PR DESCRIPTION
**Pull request by AloeareV**

### CODE REVIEW

**Comment by fluidvanadium** [comment]
im confused about the commit history on this one

**Comment by AloeareV** [comment]
> im confused about the commit history on this one

Ah! Yes, I did a `git merge -p` so there's a merge commit that only takes a few of the changes from the 'merged in' commits.

**Review by fluidvanadium — CHANGES_REQUESTED** [review]
This is great! Can you move the TransactionRecordMap implementation to transaction_records_by_id?

**Comment by fluidvanadium** [comment]
> This is great! Can you move the TransactionRecordMap implementation to transaction_records_by_id?

https://github.com/AloeareV/zingolib/pull/44

**Review by zancas — CHANGES_REQUESTED** [review]
I am seeing `integration-tests::integrations slow::timed_sync_interrupt` fail against `AloeareV:input_source`.

I am currently running the test against `dev`.   Depending on results I might loop it for 10 runs against each branch and leave that running.

We'll need to understand more about this test failure before we can land this PR.

**Review by zancas — CHANGES_REQUESTED** [review]
I *LOVE* that there's a new test here.
I prefer it to be darkside for integration-testing.   Can we port that test to darkside?

Also, I didn't check carefully..   are there unit-tests?

File: zingolib/src/wallet/traits.rs [hunk 1]
```diff
@@ -503,6 +505,13 @@ impl DomainWalletExt for SaplingDomain {
 
     type Bundle = sapling_crypto::Bundle<sapling_crypto::bundle::Authorized, Amount>;
 
+    fn get_nullifier_from_note_fvk_and_witness_position(
```
Comment (zancas) [inline zingolib/src/wallet/traits.rs:508]: Any chance of a unit test for this?

File: zingolib/src/wallet/traits.rs [hunk 2]
```diff
@@ -574,6 +580,13 @@ impl DomainWalletExt for OrchardDomain {
 
     type Bundle = orchard::bundle::Bundle<orchard::bundle::Authorized, Amount>;
 
+    fn get_nullifier_from_note_fvk_and_witness_position(
```
Comment (zancas) [inline zingolib/src/wallet/traits.rs:583]: unit test?

File: zingolib/src/wallet/transaction_record.rs [hunk 3]
```diff
@@ -110,6 +112,43 @@ impl TransactionRecord {
         }
     }
 
+    pub fn select_unspent_note_noteref_pairs_sapling(
```
Comment (zancas) [inline zingolib/src/wallet/transaction_record.rs:115]: Unit test?   doc-test?  doc-comment?

File: zingolib/src/wallet/transaction_record.rs [hunk 4]
```diff
@@ -171,6 +210,37 @@ impl TransactionRecord {
             self.total_orchard_value_spent,
         ]
     }
+    pub fn get_received_note<D>(
```
Comment (zancas) [inline zingolib/src/wallet/transaction_record.rs:213]: unit, doc-{test/comment} ?

File: zingolib/src/wallet/transactions.rs [hunk 5]
```diff
@@ -1,7 +1,55 @@
...
+use zcash_primitives::transaction::TxId;
 
+use crate::wallet::{
+    data::WitnessTrees, notes::NoteRecordIdentifier, traits::DomainWalletExt,
+    transaction_records_by_id::TransactionRecordsById,
+};
+
+use super::data::TransactionRecord;
+
+#[derive(Debug)]
```
Comment (zancas) [inline zingolib/src/wallet/transactions.rs:13]: This seems to be obsolete the type is named `TransactionRecordsById` in dev, and has moved.

File: zingolib/src/wallet/transactions/trait_inputsource.rs [hunk 6]
```diff
@@ -0,0 +1,139 @@
+use orchard::note_encryption::OrchardDomain;
```
Comment (zancas) [inline zingolib/src/wallet/transactions/trait_inputsource.rs:1]: IIRC (If I Recall Correctly) this has moved, so this feature probably needs to merge in latest dev.

**Comment by zancas** [comment]
Superceded by #935

**Pull request closed without merging**